### PR TITLE
Add attribution and text source models

### PR DESCRIPTION
**Pull request by fivetran-avinash**
**Link the issue/feature request which this PR is meant to address**
This is a PR designed to create sources to flow into sap master attribution models (0comp_code_attr, 0customer_attr, 0employee_attr, 0vendor_attr) and text models (0comp_code_text, 0company_text, 0vendor_text) created in `dbt_sap`. All of these updates will be bucketed into our initial public release of SAP packages. These are add-ons to the first initial release, as this current package is currently in a silent release state. 

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt compile
- [x] dbt run –full-refresh
- [x] dbt run
- [x] dbt test
- [N/A] dbt run –vars (if applicable)

Before marking this PR as "ready for review" the following have been applied:
- [N/A] The appropriate issue has been linked and tagged
- [N/A] You are assigned to the corresponding issue and this PR
- [x] BuildKite integration tests are passing

### Detailed Validation
Please acknowledge that the following validation checks have been performed prior to marking this PR as "ready for review":
- [x] You have validated these changes and assure this PR will address the respective Issue/Feature.
- [x] You are reasonably confident these changes will not impact any other components of this package or any dependent packages.
- [x] You have provided details below around the validation steps performed to gain confidence in these changes.
I utlilized our dbt package automation to build macros/source files and built documentations based off SAP transaction processes we went over with our product team (See the Height tickets for these tasks for more details). 

### Standard Updates
Please acknowledge that your PR contains the following standard updates:
- Package versioning has been appropriately indexed in the following locations:
    - [N/A] indexed within dbt_project.yml (this is part of the initial release process so no versioning needed)
    - [N/A] indexed within integration_tests/dbt_project.yml (ditto)
- [x] CHANGELOG has individual entries for each respective change in this PR (CHANGELOG updated to reflect initial release version)
    
- [x] README updates have been applied (if applicable)
    
- [N/A] DECISIONLOG updates have been updated (if applicable)
- [x] Appropriate yml documentation has been added (if applicable)

### dbt Docs
Please acknowledge that after the above were all completed the below were applied to your branch:
- [x] docs were regenerated (unless this PR does not include any code or yml updates)

### If you had to summarize this PR in an emoji, which would it be?
 
🇩🇪

### CODE REVIEW

File: models/stg_sap.yml [hunk 1]
```diff
@@ -0,0 +1,2158 @@
...
+        description: Number
+      - name: anz37
+        description: Number
+      - name: anz38
+        description: Number
+      - name: anz39
+        description: Number
+      - name: anz40
+        description: Number
+      - name: bet01
```
Comment (fivetran-joemarkiewicz) [inline models/stg_sap.yml]: Are these all intended to be the same description?

File: models/stg_sap.yml [hunk 2]
```diff
@@ -0,0 +1,2158 @@
...
+      - name: bet36
+        description: Wage type amount for payments
+      - name: bet37
+        description: Wage type amount for payments
+      - name: bet38
+        description: Wage type amount for payments
+      - name: bet39
+        description: Wage type amount for payments
+      - name: bet40
+        description: Wage type amount for payments
```
Comment (fivetran-joemarkiewicz) [inline models/stg_sap.yml:1541]: Are these all intended to be the same description?

File: models/stg_sap.yml [hunk 3]
```diff
@@ -0,0 +1,2158 @@
...
+      - name: ein36
+        description: Time/measurement unit 
+      - name: ein37
+        description: Time/measurement unit
+      - name: ein38
+        description: Time/measurement unit
+      - name: ein39
+        description: Time/measurement unit
+      - name: ein40
+        description: Time/measurement unit
```
Comment (fivetran-joemarkiewicz) [inline models/stg_sap.yml:1627]: Are these all intended to be the same description?

File: models/stg_sap.yml [hunk 4]
```diff
@@ -0,0 +1,2158 @@
...
+      - name: falgr
+        description: Case group
+      - name: flag1
+        description: Reserved field/unused field
+      - name: flag2
+        description: Reserved field/unused field
+      - name: flag3
+        description: Reserved field/unused field
+      - name: flag4
+        description: Reserved field/unused field
```
Comment (fivetran-joemarkiewicz) [inline models/stg_sap.yml:1639]: Are these all intended to be the same description?

File: models/stg_sap.yml [hunk 5]
```diff
@@ -0,0 +1,2158 @@
...
+      - name: ind36
+        description: Indicator for indirect valuation
+      - name: ind37
+        description: Indicator for indirect valuation
+      - name: ind38
+        description: Indicator for indirect valuation
+      - name: ind39
+        description: Indicator for indirect valuation
+      - name: ind40
+        description: Indicator for indirect valuation
```
Comment (fivetran-joemarkiewicz) [inline models/stg_sap.yml:1725]: Are these all intended to be the same description?

File: models/stg_sap.yml [hunk 6]
```diff
@@ -0,0 +1,2158 @@
...
+      - name: lga36
+        description: Wage type 
+      - name: lga37
+        description: Wage type 
+      - name: lga38
+        description: Wage type 
+      - name: lga39
+        description: Wage type 
+      - name: lga40
+        description: Wage type
```
Comment (fivetran-joemarkiewicz) [inline models/stg_sap.yml:1809]: Are these all intended to be the same description?

File: models/stg_sap.yml [hunk 7]
```diff
@@ -0,0 +1,2158 @@
...
+        description: Operation indicator for wage types
+      - name: opk37
+        description: Operation indicator for wage types
+      - name: opk38
+        description: Operation indicator for wage types 
+      - name: opk39
+        description: Operation indicator for wage types 
+      - name: opk40
+        description: Operation indicator for wage types 
+      - name: ordex
```
Comment (fivetran-joemarkiewicz) [inline models/stg_sap.yml:1890]: Are these all intended to be the same description?

File: models/stg_sap.yml [hunk 8]
```diff
@@ -0,0 +1,2158 @@
...
+      - name: rfp16
+        description: Reference personnel number
+      - name: rfp17
+        description: Reference personnel number
+      - name: rfp18
+        description: Reference personnel number 
+      - name: rfp19
+        description: Reference personnel number 
+      - name: rfp20
+        description: Reference personnel number
```
Comment (fivetran-joemarkiewicz) [inline models/stg_sap.yml:2026]: Are these all intended to be the same description?

File: README.md [hunk 9]
```diff
@@ -27,9 +27,18 @@ To use this dbt package, you must have the Fivetran **SAP** (sap.com) the respec
 - bseg
 - faglflexa
 - faglflext
+- kna1
```
Comment (fivetran-joemarkiewicz) [inline README.md]: Can the step 1 section be rephrased a bit. It sounds a bit strange when reading it over. Additionally, can we link the SAP connector that is used so customers can know exactly where they need to ensure these tables are being synced.

Reminder to also check with Edwin what the proper link for the connector is. I believe it is the HVA connector, but would prefer to confirm with him.

Comment (fivetran-joemarkiewicz) [reply README.md]: > https://fivetran.com/docs/databases/sap-erp/high-volume-agent/hva-sap-ecc-hana

File: README.md [hunk 10]
```diff
@@ -27,9 +27,18 @@ To use this dbt package, you must have the Fivetran **SAP** (sap.com) the respec
 - bseg
```
Comment (fivetran-joemarkiewicz) [inline README.md]: A few other notes on the README are as follows:
- We need to upgrade the dbt version badge to match the version in the package (greater than 1.3.0 not 1.0.0).
  - image
- You include a section on [passthrough columns](https://github.com/fivetran/dbt_sap/tree/feature/0employee-and-text-reports#passing-through-additional-fields), but we don't actually allow passthrough columns. This should be removed or updated to include passthrough column variables.
- Additionally, the optional config section should be enclosed in a collapsable. Can you add the collapsable to this section.
- Can we highlight that these staging models are intended to replicate the extractor reports (provided in our transformation package) and do not apply any renaming to the fields. If we also have a link to these extractors for reference that would be great!
- Be sure to update the package installation section to point to a dbt hub version. The first release will still be v0.1.0

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-avinash thanks for working on these updates! I do have a few comments detailed in my review below. Let me know if you have any questions when working through the comments.

Additionally, one thought I had was if we should provide variables to disable the financial models if customers do not use them. I am wondering if there is a portion of customers that may not use financial models but still want the master models 🤔 can we loop up with Edwin on this.

File: models/stg_sap.yml (hunk 1 above)
Comment (fivetran-joemarkiewicz) [reply models/stg_sap.yml]: If changes are made here, reminder to update in the src yml as well. Same comment for all the others as well.

File: README.md (hunk 10 above)
Comment (fivetran-avinash) [reply README.md]: - Version bumped
- Passthrough columns section removed
- Staging model descriptions updated
- Update the package installation section

File: README.md (hunk 9 above)
Comment (fivetran-avinash) [reply README.md]: Given Edwin's answer, would something like:

> from [Fivetran's connectors like LDP SAP Netweaver](https://fivetran.com/docs/local-data-processing/requirements/source-and-target-requirements/sap-netweaver-requirements), [HVA SAP ECC](https://fivetran.com/docs/databases/sap-erp/high-volume-agent/hva-sap-ecc-hana) or [SAP ERP on HANA](https://fivetran.com/docs/databases/sap-erp/sap-erp-hana) 

work?

File: models/stg_sap.yml (hunk 8 above)
Comment (fivetran-avinash) [reply models/stg_sap.yml:2026]: Yep!
Screenshot 2023-07-06 at 4 32 48 AM

File: models/stg_sap.yml (hunk 7 above)
Comment (fivetran-avinash) [reply models/stg_sap.yml:1890]: That appears to be case. 
Screenshot 2023-07-06 at 4 20 48 AM

File: models/stg_sap.yml (hunk 6 above)
Comment (fivetran-avinash) [reply models/stg_sap.yml:1809]: Indeed Screenshot 2023-07-06 at 4 20 48 AM

File: models/stg_sap.yml (hunk 5 above)
Comment (fivetran-avinash) [reply models/stg_sap.yml:1725]: Yep: 
Screenshot 2023-07-06 at 4 25 53 AM

File: models/stg_sap.yml (hunk 4 above)
Comment (fivetran-avinash) [reply models/stg_sap.yml:1639]: Yes 
Screenshot 2023-07-06 at 4 25 12 AM

File: models/stg_sap.yml (hunk 2 above)
Comment (fivetran-avinash) [reply models/stg_sap.yml:1541]: Yes, found a slightly more detailed description via Google but they are all the same Screenshot 2023-07-06 at 4 20 19 AM

File: models/stg_sap.yml (hunk 1 above)
Comment (fivetran-avinash) [reply models/stg_sap.yml]: Will update this to number/unit
Screenshot 2023-07-06 at 4 20 19 AM

**Review by fivetran-avinash — COMMENTED** [review]
@fivetran-joemarkiewicz Source review ready for re-review. Will follow up with Edwin about the repetitive descriptions

File: README.md (hunk 9 above)
Comment (fivetran-joemarkiewicz) [reply README.md]: Yup that is great!

Comment (fivetran-joemarkiewicz) [reply README.md]: On the same note as the transform package, I would recommend reworking this section to be something like the following.

```md
## Step 1: Prerequisites
To use this dbt package, you must have the following:
- At least one Fivetran of the following SAP connectors:
   - [LDP SAP Netweaver](https://fivetran.com/docs/local-data-processing/requirements/source-and-target-requirements/sap-netweaver-requirements)
   - [HVA SAP ECC](https://fivetran.com/docs/databases/sap-erp/high-volume-agent/hva-sap-ecc-hana)
   - [SAP ERP on HANA](https://fivetran.com/docs/databases/sap-erp/sap-erp-hana) 
-  Within the connector, syncing the following respective tables into your destination:
  - bkpf
  - bseg
  - faglflexa
  - faglflext
  - kna1
  - lfa1
  - mara
  - pa0000
  - pa0001
  - pa0007
  - pa0008
  - pa0031
  - ska1
  - t001
  - t503
  - t880
- A **BigQuery**, **Snowflake**, **Redshift**, **PostgreSQL**, **Databricks** destination.
```

File: README.md (hunk 10 above)
Comment (fivetran-joemarkiewicz) [reply README.md]: Thanks! A few other small things I noticed that should be updated in the README that I noticed after this last update:
- Don't forget to add the databricks dispatch in the same location that you have in the transform README.
- The https://github.com/fivetran/dbt_sap_source/blob/feature/employee-text-reports/README.md#change-the-source-table-references section does not seem to be included in the collapsable section
- In the change-the-source-table-references you reference SAP2 sources. What are these?

File: README.md (hunk 9 above)
Comment (fivetran-joemarkiewicz) [reply README.md]: Main point here is to have this section be consistent between both package READMEs.

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-avinash this is looking great!! I only have a few unresolved comments that need to be addressed before approving. Let me know if you have any questions!

File: models/stg_sap.yml (hunk 3 above)
Comment (fivetran-avinash) [reply models/stg_sap.yml:1627]: Yes, see other screenshots with similar comments.

File: README.md (hunk 9 above)
Comment (fivetran-avinash) [reply README.md]: Updated

File: README.md (hunk 10 above)
Comment (fivetran-avinash) [reply README.md]: - Databricks dispatch updated
- </details> moved to include section in collapsible
- Uhh, good question. I think that's from the oldest version of this model, not sure what that's referencing. Removed.

**Review by fivetran-avinash — COMMENTED** [review]
@fivetran-joemarkiewicz All necessary changes applied. One last go-around!

File: README.md [hunk 11]
```diff
@@ -21,23 +23,45 @@
...
+  - pa0000
+  - pa0001
+  - pa0007
+  - pa0008
+  - pa0031
+  - ska1
+  - t001
+  - t503
+  - t880
+- A **BigQuery**, **Snowflake**, **Redshift**, **PostgreSQL**, **Databricks** destination.
```
Comment (fivetran-joemarkiewicz) [inline README.md]: The bullets don't seem to be consistent in the rendered preview. I think the second set of indented bullets need to be adjusted.
image

```suggestion
- At least one Fivetran of the following SAP connectors:
   - [LDP SAP Netweaver](https://fivetran.com/docs/local-data-processing/requirements/source-and-target-requirements/sap-netweaver-requirements)
   - [HVA SAP ECC](https://fivetran.com/docs/databases/sap-erp/high-volume-agent/hva-sap-ecc-hana)
   - [SAP ERP on HANA](https://fivetran.com/docs/databases/sap-erp/sap-erp-hana) 
-  Within the connector, syncing the following respective tables into your destination:
   - bkpf
   - bseg
   - faglflexa
   - faglflext
   - kna1
   - lfa1
   - mara
   - pa0000
   - pa0001
   - pa0007
   - pa0008
   - pa0031
   - ska1
   - t001
   - t503
   - t880
- A **BigQuery**, **Snowflake**, **Redshift**, **PostgreSQL**, **Databricks** destination.
```

**Review by fivetran-joemarkiewicz — APPROVED** [review]
@fivetran-avinash there was just one edit in the README for bullet indent consistency. Once that is applied, this will be good to merge!

File: README.md (hunk 11 above)
Comment (fivetran-avinash) [reply README.md]: You apparently have to space when you're indenting a bullet point =P. This is fixed on both source and transform packages.

**Pull request merged**